website/docs/r/vapp_vm.html.markdown

```diff
@@ -239,8 +239,8 @@ The following arguments are supported:
 * `catalog_name` - (Optional; *v2.9+*) The catalog name in which to find the given vApp Template or media for `boot_image`.
 * `template_name` - (Optional; *v2.9+*) The name of the vApp Template to use
 * `vm_name_in_template` - (Optional; *v2.9+*) The name of the VM in vApp Template to use. For cases when vApp template has more than one VM.
-* `memory` - (Optional) The amount of RAM (in MB) to allocate to the VM
-* `cpus` - (Optional) The number of virtual CPUs to allocate to the VM. Socket count is a result of: virtual logical processors/cores per socket. The default is 1
+* `memory` - (Optional) The amount of RAM (in MB) to allocate to the VM. If `memory_hot_add_enabled` is true, then memory will be increased without VM restart.
+* `cpus` - (Optional) The number of virtual CPUs to allocate to the VM. Socket count is a result of: virtual logical processors/cores per socket. The default is 1. If `cpu_hot_add_enabled` is true, then cpus will be increased without VM restart.
```

Comment:
That's a great statement, but it doesn't work like that. If I change it, VM is restarted. What's the plan to approach this?

Reply:
please give me example, I don't follow.

Reply:
What I mean is that I changed `cpus` count with `cpu_hot_add_enabled = true` and it still restarted the VM.

Do I understand right, that this functionality was not there yet in this PR Yesterday? But Today it should start working, because you committed:
https://github.com/terraform-providers/terraform-provider-vcd/commit/cb7d5dc30fe1640377f4540b3c22ab0172b1dca3
>  * `resource/vcd_vapp_vm` allows change `network` parameters without VM power off [GH-536]

Reply:
I think you miss one thing. To change `cpu_hot_add_enabled` you need VM power off, when `cpu_hot_add_enabled` enabled then you can change `cpus` withouth power off.

Reply:
This functionality was already when you tested. I added in the morning possibility to change `network` fields without VM power off.

Reply:
> I think you miss one thing. To change cpu_hot_add_enabled you need VM power off, when cpu_hot_add_enabled enabled then you can change cpus withouth power off.

I did miss that! Maybe we should add a sentence to docs to parameters which reboot the VM?
`Changing this parameter requires VM reboot`

So, I tried increasing `cpus` and `memory` and the VM was not rebooted, yay!

Reply:
> This functionality was already when you tested. I pushed in the morning possibility to change network fields without VM power off.

Can all fields of `network` block be changed without reboot anymore? Or are there some limitations to be aware of?

Reply:
I tried and all them can be changed, except adapter type - which already mentioned in documention.

Reply:
I have experimented:

1. Removing `network` block - restart (expected)

2. Adding `network` block - hot add (expected)

3. But then I switched `is_primary` from `true` to `false` and **it was not noticed**:

```
  # 2.10.0 - hot add network
  network {
    type               = "org"
    name               = vcd_vapp_org_network.demo_vapp_net_org.org_network_name
    ip_allocation_mode = "POOL"
    is_primary         = false
  }
```
Output:
```
$ terraform apply
vcd_external_network.demo_ext_net: Refreshing state... [id=urn:vcloud:network:30759524-ff11-4236-9579-8b2b1f18c0f7]
vcd_org.demo_org: Refreshing state... [id=urn:vcloud:org:992e69cd-fd5e-4815-862a-d0c010825ae7]
vcd_org_vdc.demo_vdc: Refreshing state... [id=urn:vcloud:vdc:1e4a69c8-dff5-4ce7-9a23-c36605cc7743]
vcd_edgegateway.demo_gw: Refreshing state... [id=urn:vcloud:gateway:9a050d14-2afb-4c0e-8b67-178aff85d5d6]
vcd_network_routed.demo_routed_net_web: Refreshing state... [id=urn:vcloud:network:612d27d9-493d-479b-a8e9-ccdbebdc29b7]
vcd_nsxv_firewall_rule.demo_nsxv_fw_rule: Refreshing state... [id=131078]
vcd_vapp.demo_vapp: Refreshing state... [id=urn:vcloud:vapp:0983cdbb-61b2-460d-8e63-cdfeec8dbb85]
vcd_vapp_org_network.demo_vapp_net_org: Refreshing state... [id=urn:vcloud:network:57eff952-7363-4d4c-b246-3fa0edf70fb4]
vcd_vapp_network.demo_vapp_net_w_org: Refreshing state... [id=urn:vcloud:network:1e6a60ad-3195-42e8-b9e7-58d39330a8ad]
vcd_vapp_static_routing.vapp_demo_static_routing1: Refreshing state... [id=urn:vcloud:network:1e6a60ad-3195-42e8-b9e7-58d39330a8ad]
vcd_vapp_firewall_rules.vapp_fw: Refreshing state... [id=urn:vcloud:network:1e6a60ad-3195-42e8-b9e7-58d39330a8ad]
vcd_vapp_firewall_rules.vapp_org_fw: Refreshing state... [id=urn:vcloud:network:57eff952-7363-4d4c-b246-3fa0edf70fb4]
vcd_vapp_vm.demo_vm_photon: Refreshing state... [id=urn:vcloud:vm:cdfccf69-ebbc-4918-b8c7-ad141776bedf]
vcd_vapp_vm.demo_vm_wordpress: Refreshing state... [id=urn:vcloud:vm:a60a0c5f-0498-4610-8f48-27c7c9de8f5b]
vcd_nsxv_dnat.demo_dnat_wordpress_https1: Refreshing state... [id=196609]
vcd_vapp_nat_rules.vapp_nat: Refreshing state... [id=urn:vcloud:network:1e6a60ad-3195-42e8-b9e7-58d39330a8ad]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

Please try reproducing.

CC: @Didainius

Reply:
"is_primary": {
					Default:  false,
					Optional: true,
					// By default if the value is omitted it will report schema change
					// on every terraform operation. The below function
					// suppresses such cases "" => "false" when applying.
					DiffSuppressFunc: falseBoolSuppress(),
					Type:             schema.TypeBool,
					Description:      "Set to true if network interface should be primary. First network card in the list will be primary by default",
				},

Reply:
This PR didn't have impact to behavior here. The reason it works so - one  adapter has to have "is_primary" flag  anyway(radio button in UI - one bubble has to be checked). So if you remove it from one - it still stays primary. You have to explicitly add `is_primary` to some other NIC.

Reply:
Thanks. @Didainius also added:
> No-op if you don't set "true" on some other NIC
> One cannot disable/switch off is primary for NIC unless some other NIC is set so